### ISSUE TITLE
CC3200 I²C at 100KHz doesn't work

### ISSUE THREAD

**Issue by rei-vilo** [energia/cc3200-core#7]
From https://github.com/energia/Energia/issues/941

**Comment by rei-vilo** [comment]
Also #2 about I²C.

**Comment by robertinant** [comment]
I was able to reproduce this. CCS was indeed fine with 100KHz.
I also found the issue. I2CTransact() was checking for master busy. If it is then I would return. This is wrong. This snip was taken from the original example way back when. It now has been corrected in that same example and there is no longer a check for master busy. If I insert that same line into the function in CCS, it also fails at 100KHz confirming the fix.
This is the patch. You will still need to set fastmode to false as you have done previously. I currently plan on keeping it at 400KHz by default. I might enhance the library at some point to allow configuring it.

`--- a/libraries/Wire/Wire.cpp
+++ b/libraries/Wire/Wire.cpp
@@ -69,9 +69,7 @@ uint8_t TwoWire::I2CTransact(unsigned long ulCmd)
        MAP_I2CMasterControl(I2C_BASE, ulCmd);
 
        while((MAP_I2CMasterIntStatusEx(I2C_BASE, false)
-                       & (I2C_INT_MASTER | I2C_MRIS_CLKTOUT)) == 0) {
-               if(!MAP_I2CMasterBusy(I2C_BASE)) return -1;
-       }
+                       & (I2C_INT_MASTER | I2C_MRIS_CLKTOUT)) == 0) { }
 
        /* Check for any errors in transfer */
        if(MAP_I2CMasterErr(I2C_BASE) != I2C_MASTER_ERR_NONE) {`
